src/ExamplesGenerator/AsciiDocGenerator.cs

```diff
@@ -28,8 +30,17 @@ static AsciiDocGenerator()
 				.WithChangedOption(TabSize, CSharp, 4);
 		}
 
-		public static void GenerateExampleAsciiDoc(IEnumerable<ImplementedExample> examples, string branchName)
+		public static void GenerateExampleAsciiDoc(IEnumerable<ImplementedExample> examples, string branchName, string path)
 		{
+			var referencePages = AsciiDocParser.ParsePages(path);
+
+			string GetReferencePage(ImplementedExample e)
+			{
+				return referencePages.Select(p => new { Hashes = p.Examples.Select(e => e.Hash), File = p.Name })
+					.Where(o => o.Hashes.Contains(e.Hash))
+					.Select(h => h.File).FirstOrDefault();
+			}
```

Comment:
I think it would be better to write the reference page file into the generate example method e.g. as a comment on the method, and parse it out when creating instances of `ImplementedExample`

Reply:
Potentially, I think having the lookup is also an additional useful bookeeping step.